packages/terra-worklist-data-grid/src/subcomponents/RowSelectionCell.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import '../_elementPolyfill';
+import { injectIntl } from 'react-intl';
+import ThemeContext from 'terra-theme-context';
+import styles from '../WorklistDataGrid.module.scss';
+import Cell from './Cell';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * String identifier of the row in which the Cell will be rendered.
+   */
+  rowId: PropTypes.string.isRequired,
+
+  /**
+   * String identifier of the column in which the Cell will be rendered.
+   */
+  columnId: PropTypes.string.isRequired,
+  coordinates: PropTypes.shape({
+    row: PropTypes.number,
+    col: PropTypes.number,
+  }),
+
+  /**
+   * Boolean value to indicate if the cell can accept focus.
+   */
+  acceptsFocus: PropTypes.bool,
+
+  /**
+   * Boolean indicating whether the Cell is actively selected.
+   */
+  isSelected: PropTypes.bool,
+
+  /**
+   * String that labels the cell for accessibility.
+   */
+  ariaLabel: PropTypes.string,
+
+  onCellSelectionChange: PropTypes.func,
+  onMoveCellFocus: PropTypes.func,
+  isNavigationEnabled: PropTypes.bool,
+  isRowSelectionModeEnabled: PropTypes.bool,
+
+  /**
+   * @private
+   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+function RowSelectionCell(props) {
+  const {
+    rowId,
+    columnId,
+    coordinates,
+    acceptsFocus,
+    isSelected,
+    ariaLabel,
+    onCellSelectionChange,
+    onMoveCellFocus,
+    isNavigationEnabled,
+    isRowSelectionModeEnabled,
+    intl,
+    ...customProps
+  } = props;
+
+  const theme = useContext(ThemeContext);
+  const rowLabel = intl.formatMessage({ id: 'Terra.worklist-data-grid.row-index' }, { row: coordinates.row });
+
+  return (
+    <Cell
+      rowId={rowId}
+      columnId={columnId}
+      coordinates={coordinates}
+      acceptsFocus={false}
+      isSelected={isSelected}
+      isRowSelectionCell
+      onCellSelectionChange={onCellSelectionChange}
+      onMoveCellFocus={onMoveCellFocus}
+      isRowSelectionModeEnabled={isRowSelectionModeEnabled}
+      className={cx(['selectable', theme.className])}
+      isNavigationEnabled={isNavigationEnabled}
+      rowsLength={customProps.rowsLength}
+      columnsLength={customProps.columnsLength}
+    >
+      <input
+        type="checkbox"
+        aria-label={ariaLabel || rowLabel}
```

Comment:
This checkbox is just reading as "checkbox". At the very least it should read "Select Row [row label]"